scripts/populate_cel_table.py

```diff
@@ -0,0 +1,128 @@
+import csv
+import itertools
+import os
+import sys
+from collections.abc import Iterator, Mapping
+from itertools import count
+
+import sqlalchemy
+
+ASSETS_DIR = os.getenv('WAZO_BENCHMARK_ASSETS_DIR', 'assets')
+DATABASE_URI = os.getenv('WAZO_BENCHMARK_DATABASE_URI', 'postgresql://localhost:5432/asterisk')
+MAX_GENERATED_CELS = int(os.getenv('WAZO_BENCHMARK_MAX_CELS') or 100_000_000)
+GENERATED_CELS_FILENAME = os.getenv('WAZO_BENCHMARK_GENERATED_CELS_FILENAME', f'{MAX_GENERATED_CELS}_cels.csv')
+
+CEL_COLUMNS = (
+    'id',
+    'eventtype',
+    'eventtime',
+    'userdeftype',
+    'cid_name',
+    'cid_num',
+    'cid_ani',
+    'cid_rdnis',
+    'cid_dnid',
+    'exten',
+    'context',
+    'channame',
+    'appname',
+    'appdata',
+    'amaflags',
+    'accountcode',
+    'peeraccount',
+    'uniqueid',
+    'linkedid',
+    'userfield',
+    'peer',
+    'call_log_id',
+    'extra',
+)
+
+
+def log(*args):
+    print(*args, file=sys.stderr)
+
+
+def generate_uniqueid(linkedid):
+    (head, tail) = linkedid.split('.')
+    yield linkedid
+    current_tail = int(tail) + 1
+    while True:
+        yield f'{head}.{current_tail}'
+        current_tail += 1
+
+
+def generate_linkedid(initial='0.0'):
+    (head, tail) = initial.split('.')
+    yield initial
+    current_head = int(head) + 1
+    while True:
+        yield f'{current_head}.{tail}'
+        current_head += 1
+
+
+def generate_new_cel_sequence(cel_indexer, new_linkedid, initiator_cels):
+    linkedid = new_linkedid
+    uniqueid_gen = generate_uniqueid(linkedid)
+    initator_linkedid = next(cel['linkedid'] for cel in initiator_cels)
+    uniqueid_map = {initator_linkedid: linkedid}
+    for cel in initiator_cels:
+        new_cel = dict(cel)
+        new_cel['id'] = next(cel_indexer)
+        new_cel['linkedid'] = linkedid
+        if cel['uniqueid'] in uniqueid_map:
+            new_cel['uniqueid'] = uniqueid_map[cel['uniqueid']]
+        else:
+            new_cel['uniqueid'] = uniqueid_map[cel['uniqueid']] = next(uniqueid_gen)
+        new_cel['call_log_id'] = None
+        yield new_cel
+
+
+def generate_cel_sequences(base_cels):
+    cel_indexer = count(1)
```

Comment:
I have an integrity error when the script reaches existing cel IDs.
You could either set the start at the value in `select last_value from cel_id_seq` or omit the id field and allow postgresql to assign the next available ID automatically

Reply:
No problem not setting the `id` column.